test/test_base.py

```diff
@@ -42,6 +46,18 @@ def setup_logging(self):
         self.log = logging.getLogger("test")
         self.log.info("Logging started")
 
+    def print_and_assert_msgs(self, message):
+        if (self.log_expected_msgs or self.log_unexpected_msgs) and assert_log(
+            message, self.log_expected_msgs, self.log_unexpected_msgs
+        ):
+            self.log_msg_assertions_passed = True
+        print(message)
+
+    def assert_log_msgs_passed(self):
+        assert (
+            self.log_msg_assertions_passed
+        ), f"Log assertion failed: {self.log_expected_msgs}, {self.log_unexpected_msgs}"
```

Comment:
since unexpected messages get their own error and log message on L213 I think you can be more explicit here:

```suggestion
        ), f"Log assertion failed. Expected message not found: {self.log_expected_msgs}"
```